dask_expr/expr.py

```diff
@@ -398,6 +398,17 @@ def value_counts(self, sort=None, ascending=False, dropna=True, normalize=False)
             raise NotImplementedError("value_counts not implemented for DataFrame")
         return ValueCounts(self, sort, ascending, dropna, normalize)
 
+    def memory_usage(self, deep=False, index=no_default):
+        if is_index_like(self._meta):
+            if index is not no_default:
+                raise TypeError(
+                    "index key-word is not supported when calculating memory_usage of index"
+                )
+            return MemoryUsage(self, deep=deep)
+        if index is no_default:
+            index = True
+        return MemoryUsage(self, deep=deep, _index=index)
```

Comment:
An alternative is to stop putting these methods on `Expr` and instead put them on `DataFrame/Series/Index`.  This makes the `Expr` class a little less useful on its own, but maybe that's ok because it's rarely used on its own.  

I propose that we use this reduction as an example, and see how this would look if we put the methods on the DataFrame/Series/Index classes directly and see if that feels cleaner.

Thoughts?

Reply:
Added it to Series/Index/Frame, this is probably the cleanest we can come up with with regards to if/else stuff. Thoughts?